pyproject.toml

```diff
@@ -1,12 +1,18 @@
 [project]
-requires-python = ">=3.11, <3.13"
+name = "qpc"
+version = "1.8.0"
```

Comment:
Do we need this `version` specification? This adds another place to remember to update every time we bump versions to release. Or can we remove the `version` down under `[tool.poetry]` if we prefer/need this one under `[project]`?

Ditto for `name` since we have to patch that downstream.

Reply:
poetry vs 3.12 setuptools fighting each other :( 

with name and version only in `[project]`:

```
$ poetry run qpc

The Poetry configuration is invalid:
  - The fields ['name', 'version'] are required in package mode.
```

with name and version only in `[tool.poetry]`:

```
$ python setup.py
Traceback (most recent call last):
...
ValueError: invalid pyproject.toml config: `project`.
configuration error: `project` must contain ['name'] properties
```

Reply:
I don't know by heart, but I know specfiles in RHEL do support PEP-517/518 compliant projects (aka, without setup.py). Maybe this is a good time to get rid of `setup.py` for good?

Reply:
Understood. I was afraid that might happen. 😞 

In that case, we should add a check somewhere to ensure that these two fields stay in sync. Maybe add a make target that we can run locally and in a GitHub action to ensure that we never update one and forget the other. We could use `tomllib` to read the file and check these two fields, assuming there's not a better way to parse it.

Reply:
Only thing holding us back is that we would also need to change downstream patching mechanism (maybe we could also just get rid of qpc name and be done with it)

Reply:
Orrrrr maybe we consider what @bruno-fs said. 🤔 

(Beat me by seconds?? gotta go fast. 🦔)

Reply:
> In that case, we should add a check somewhere to ensure that these two fields stay in sync.

I think I did that in chaski with some third party tool 🤔... lemme check

Reply:
Found it.
https://github.com/quipucords/chaski/blob/main/pyproject.toml#L31C1-L34C1

the tool accepts pointing python files as well

https://commitizen-tools.github.io/commitizen/commands/bump/#version_files

Reply:
Doing a quick dig, I found these macros

```
%generate_buildrequires
%pyproject_buildrequires
%pyproject_wheel
%install
%pyproject_install
%pyproject_save_files
```
source: https://src.fedoraproject.org/rpms/pyproject-rpm-macros

However, if you decide to go this path, I'd invert the order of implementation and try to make it work downstream FIRST as the availability of dependencies is more scarce there.

TIP: we **_DON'T NEED POETRY_**. Just `poetry-core` as build dependency should suffice (if the macro does not "magically" solve it)

Reply:
example implementation: https://url.corp.redhat.com/13ecab7
(cryptography doesn't rely on poetry but it is a package that doesn't have setup.py)

Reply:
Just to keep everyone on the loop, in order to use pyproject template tags (and poetry-core) on RHEL8 we would need to build them all from scratch. Let's postpone removing setup.py as this "tech debt" is clearly big enough to deserve its own Jira ticket.